kivymd/uix/tab.py

```diff
@@ -720,40 +788,81 @@ class MDTabsBase(Widget):
     and defaults to `None`.
     """
 
+    def _get_label_font_style(self):
+        if self.tab_label:
+            return self.tab_label.font_style
+
+    def _set_label_font_style(self, value):
+        if self.tab_label:
+            if value in theme_font_styles:
+                self.tab_label.font_style = value
+            else:
+                raise ValueError(
+                    "tab_label_font_style:\n\t"
+                    "font_style not found in theme_font_styles\n\t"
+                    f"font_style = {value}"
+                )
+        else:
+            Clock.schedule_once(lambda x: self._set_label_font_style(value))
+            return True
+
+    tab_label_font_style = AliasProperty(
+        _get_label_font_style,
+        _set_label_font_style,
+        cache=True,
+    )
+    """
+    :attr:`tab_label_font_style` is an :class:`~kivy.properties.AliasProperty`
+    that behavies similar to an :class:`~kivy.properties.OptionProperty`.
+
+    This property's behavior allows the developer to use any new label style
+    registered to the app.
+
+    This property will affect the Tab's Title Label widget.
+
+    """
+
     def __init__(self, **kwargs):
         self.tab_label = MDTabsLabel(tab=self)
         super().__init__(**kwargs)
-        self.font_style = "Button"
         self.bind(
             icon=self._update_text,
             title=self._update_text,
             title_icon_mode=self._update_text,
+            text=self.update_label_text,
+            tab_label_text=self.update_label_text,
         )
-        self._update_text()
+        Clock.schedule_once(
+            self._update_text
+        )  # This will ensure the text is correct
 
     def _update_text(self, *args):
-        x = False
+        # ensures that the title is in capital letters.
         if self.title:
-            self.title = self.title.upper()
+            if self.title != self.title.upper():
```

Comment:
PLEASE - do not hard code the label title to Upper case.

This is a deal-breaker. There is NO specification in material.io that says that Tab labels should be all caps.

While you might think it is good practice and I might agree with you - this MUST be a matter for the app designer.

Reply:
Well, the material io requieres that all labels must be clear, and. As far as I know, common practice to this is to use capital letters, since they have the most serif and use a bigger space to be drawn. But there's no problem, it can be removed

Reply:
yet, i just changed it to be optional.

Reply:
> As far as I know, common practice to this is to use capital letters, since they have the most serif and use a bigger space to be drawn

Thanks for making it optional.

Hard coding "good practice" is really not a good practice. And I do not necessarily agree with you. I use san-serif fonts (and Roboto is also san-serif) so that the question of the size of the serif has no relevance.

And if you look at from the point of view of psychology - all-caps words are harder for humans to read because we actually read by pattern recognition on the word and not by reading the letters.